robotframework-ls/src/robotframework_ls/impl/flow_explorer_model_builder.py

```diff
@@ -1,9 +1,53 @@
+from contextlib import contextmanager
 from robotframework_ls.impl.protocols import ICompletionContext, IKeywordFound, INode
 from typing import List, Any, Dict, Tuple
 import os
 from robocorp_ls_core.basic import isinstance_name
 
 
+class _KeywordRecursionStack:
+    _stack: Dict
+
+    def __init__(self) -> None:
+        self._stack = {}
+
+    def has(self, keyword_name: str) -> bool:
```

Comment:
The pythonic thing here would probably be changing `has` to `__contains__` and then doing `if keyword_name in recursion_stack`.